flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -307,6 +316,7 @@ public PipelinedSubpartitionView createReadView(BufferAvailabilityListener avail
 
 			readView = new PipelinedSubpartitionView(this, availabilityListener);
 			notifyDataAvailable = !buffers.isEmpty();
+			dataAvailableNotified = notifyDataAvailable;
```

Comment:
Refer to https://github.com/apache/flink/pull/11786/files#r410367542, we can remove `dataAvailableNotified = notifyDataAvailable;`